lib/resources/postgres.rb

```diff
@@ -33,32 +41,47 @@ def initialize
         # warning in version_from_dir. We should determine which case
         # is more common and only warn in the less common case.
         #
-        version = if inspec.directory('/var/lib/pgsql/data').exist?
-                    warn 'Found /var/lib/pgsql/data. Assuming postgresql install uses un-versioned directories.'
-                    nil
-                  else
-                    version_from_dir('/var/lib/pgsql/')
-                  end
+        if !version_command.nil?
+          @version = version_command.to_s
+        elsif
+          inspec.directory('/var/lib/pgsql/data').exist?
+          warn 'Found /var/lib/pgsql/data. Assuming postgresql install uses
+                un-versioned directories.'
+          return @version = nil
```

Comment:
You can't call `return` here - it will cause the method to exit and the resource will not be properly initialized.